lib/vwo/core/variation_decider.rb

```diff
@@ -178,6 +224,90 @@ def get_variation_allotted(user_id, campaign)
         end
       end
 
+      private
+
+      # Evaluate all the variations in the campaign to find
+      #
+      # @param[String]  :user_id      The unique key assigned to User
+      # @param[Hash]    :campaign     Campaign hash for Unique campaign key
+      #
+      # @return[Hash]
+
+      def evaluate_Whitelisting(user_id, campaign, campaign_key, variation_targeting_variables )
+        variation_targeting_variables = variation_targeting_variables.merge({ vwo_user_id: user_id })
+        targeted_variations = []
+
+        campaign['variations'].each do |variation|
+          segments = get_segments(variation)
+          is_valid_segments = valid_value?(segments)
+          if is_valid_segments
+            if @segment_evaluator.evaluate(campaign_key, user_id, segments, variation_targeting_variables)
+              targeted_variations.push(variation)
+            end
+          else
+            @logger.log(
+              LogLevelEnum::INFO,
+              format(
+                LogMessageEnum::InfoMessages::SKIPPING_PRE_SEGMENTATION,
+                file: FILE,
+                campaign_key: campaign_key,
+                user_id: user_id,
+                api_name: ApiMethods::GET_FEATURE_VARIABLE_VALUE,
+                varaition: variation['name']
+              )
+            )
+          end
+        end
+
+        if targeted_variations.length() > 1
```

Comment:
if targeted_variations.any?

Reply:
need to check if it have more than 1 element, not sure .any would ensure that?